Directory.Packages.props

```diff
@@ -19,19 +19,19 @@
     -->
     <!-- We are using 4.8, which introduces .NET 8 and supports VS 2022 17.8 -->
     <PackageVersion Include="Microsoft.CodeAnalysis.CSharp.Workspaces" Version="4.8" />
-    <PackageVersion Include="Microsoft.CodeAnalysis.CSharp" Version="4.8" />
+    <PackageVersion Include="Microsoft.CodeAnalysis.CSharp" Version="4.14.0" />
```

Comment:
We want to pin this to 4.8: this is the first version that introduces .NET 8 and supports VS2022 (17.8). We won't bump it until .NET 8 is EOL 10-Nov, 2026